src/main/java/uk/gov/ida/eidas/bridge/resources/VerifyAuthnRequestResource.java

```
@@ -54,9 +59,23 @@ public Response receiveAuthnRequest(
 
         String authnRequestId = authnRequest.getID();
 
+        RequestedAuthnContext requestedAuthnContext = authnRequest.getRequestedAuthnContext();
+        List<AuthnContextClassRef> authnContextClassRefs = requestedAuthnContext.getAuthnContextClassRefs();
+        AuthnContext lowestAuthnContext = null;
+        AuthnContextFactory authnContextFactory = new AuthnContextFactory();
+        for(AuthnContextClassRef authnContextClassRef : authnContextClassRefs) {
```

At the moment when there's 0 authnContextClassRefs lowestAuthnContext will end up as null. Should probably throw an error in that case.

Also something like
```
lowest = authnContextClassRefs.stream()
    .map(x -> authnContextFactory.authnContextForLevelOfAssurance(x.getAuthnContextClassRef())
    .min(AuthContext::compare)
    .orElseThrow(new SomeException())
```

(untested)

Would be a more readable way of doing this in my opinion.